src/psyclone/domain/nemo/transformations/nemo_arrayrange2loop_trans.py

```diff
@@ -377,17 +385,20 @@ def validate(self, node, options=None):
 
         # Is the Range node the outermost Range (as if not, the
         # transformation would be invalid)?
-        if any(isinstance(node, Range) for node in
-               node.parent.children[node.position+1:]):
-            raise TransformationError(
-                "Error in NemoArrayRange2LoopTrans transformation. This "
-                "transformation can only be applied to the outermost "
-                "Range.")
+        after_node = False
+        for index_child in node.parent.indices:
```

Comment:
How about:

    for index_child in node.parent.indices[node.parent.indices.index(node)+1:]

and then you don't need `after_node`.

Reply:
Or even:

    if any(isinstance(idx_child, Range) for idx_child in node.parent.indices[node.parent.indices.index(node)+1:]):

Reply:
Oh yeah, I was overcomplicating things, I went with the first suggestion.